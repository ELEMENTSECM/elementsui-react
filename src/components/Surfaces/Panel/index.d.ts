import * as React from 'react';

export interface PanelProps {
	/**
	 * Panel open
	 */
	isOpen?: boolean;
	/**
	 * Panel on dismissed
	 */
	onDismissed?: (...args: any[]) => any;
	/**
	 * Panel header text
	 */
	headerText?: string;
	/**
	 * Panel closebutton aria label
	 */
	closebuttonAriaLabel?: string;
	/**
	 * Panel footer content
	 */
	onRenderFooterContent?: (...args: any[]) => any;
	/**
	 * Panel hasCloseButton
	 */
	hasCloseButton?: boolean;
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const Panel: React.SFC<PanelProps>;

export default Panel;
