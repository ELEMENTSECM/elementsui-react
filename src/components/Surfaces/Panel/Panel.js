import React from 'react';
import PropTypes from 'prop-types';
import { Panel as UIFabPanel } from 'office-ui-fabric-react/lib/Panel';

/**Panel example */
function Panel({
	headerText,
	element,
	onRenderFooterContent,
	isOpen,
	onDismissed,
	hasCloseButton,
	closebuttonAriaLabel
}) {
	return (
		<UIFabPanel
			headerText={headerText}
			isOpen={isOpen}
			onDismissed={onDismissed}
			hasCloseButton={hasCloseButton}
			closebuttonAriaLabel={closebuttonAriaLabel}
			onRenderFooterContent={onRenderFooterContent}>
			{element}
		</UIFabPanel>
	);
}

Panel.propTypes = {
	/**Panel open */
	isOpen: PropTypes.bool,
	/**Panel on dismissed */
	onDismissed: PropTypes.func,
	/**Panel header text */
	headerText: PropTypes.string,
	/**Panel closebutton aria label */
	closebuttonAriaLabel: PropTypes.string,
	/**Panel footer content */
	onRenderFooterContent: PropTypes.func,
	/**Panel hasCloseButton */
	hasCloseButton: PropTypes.bool
};

export default Panel;
