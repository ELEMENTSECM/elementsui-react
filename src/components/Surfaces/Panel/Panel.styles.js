export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Panel',
			{
				// background: props.theme.palette.themePrimary
				// place your styles here
			},
			className
		]
	};
};
