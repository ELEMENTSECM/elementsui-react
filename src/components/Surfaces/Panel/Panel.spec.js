import React from 'react';
import renderer from 'react-test-renderer';
import Panel from './Panel';

describe('Panel', () => {
	test('displays  `Foo` as headerText', () => {
		const tree = renderer.create(<Panel id="defaultBtn" headerText="Foo" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
