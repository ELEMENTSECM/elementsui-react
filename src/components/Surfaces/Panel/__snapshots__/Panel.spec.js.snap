// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel displays  \`Foo\` as headerText 1`] = `null`;
